internal/armada/repository/job.go

```diff
@@ -282,19 +290,27 @@ func (repo *RedisJobRepository) TryLeaseJobs(clusterId string, queue string, job
 	return leasedJobs, nil
 }
 
-func (repo *RedisJobRepository) GetJobsByIds(ids []string) ([]*api.Job, error) {
+// Returns existing jobs by Id
+// If an Id is supplied that no longer exists, that job will simply be omitted from the result.
+// No error will be thrown for missing jobs
+func (repo *RedisJobRepository) GetExistingJobsByIds(ids []string) ([]*api.Job, error) {
 	pipe := repo.db.Pipeline()
 	var cmds []*redis.StringCmd
 	for _, id := range ids {
 		cmds = append(cmds, pipe.Get(jobObjectPrefix+id))
 	}
-	_, e := pipe.Exec()
-	if e != nil {
-		return nil, e
-	}
+	_, _ = pipe.Exec() // ignoring error here as it will be part of individual commands
 
 	var jobs []*api.Job
-	for _, cmd := range cmds {
+	for index, cmd := range cmds {
+		_, e := cmd.Result()
+		if e != nil {
+			if e == redis.Nil {
+				log.Warnf("No job found with with job id %s", ids[index])
```

Comment:
Curious how many of these messages will we get, not sure if I would consider it warning, its expected.